### PR TITLE
Set helpers #isEqual, adds some test coverage

### PR DESCRIPTION
**Pull request by mrflip**
Addresses #172 

Implementation is that two sets are `isEqual` if they are the same object, or have the same size and B has every element of A.

I did this by duplicating the entire `isSubset` method and changing the relevant line, rather than first testing for size equality and then delegating to `isSubset` -- saving operations at the cost of code complexity.

Also have added some test coverage in there. They are largely documentary in that they cover edge cases unlikely to crop up in the current implementation, but which a naïve outsider might like reassurance about (eg that intersecting with an empty set makes an empty set, etc)

Once I've addressed your comments, I can do the same to everything else with isSubset and for other tests.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,9 @@
 # Changelog
 
+## 0.39.3
```
Comment (Yomguithereal) [inline CHANGELOG.md:3]: Should be `0.40.0` since we are adding a feature. I also usually add `(provisional)` at the right so that people know until the version is actually released.

File: test/set.js [hunk 2]
```diff
@@ -11,23 +11,93 @@ describe('Set functions', function() {
   describe('#.intersection', function() {
 
     it('should properly compute the intersection of two sets.', function() {
-      var A = new Set([1, 2, 3]),
-          B = new Set([2, 3, 4]);
+      var Exemplar = new Set([2, 3, 1]);
```
Comment (Yomguithereal) [inline test/set.js:14]: Those variables should not be capitalized.

File: test/set.js [hunk 3]
```diff
@@ -11,23 +11,93 @@ describe('Set functions', function() {
...
+      var Equiv    = new Set([1, 2, 3]);
+      var Missing1 = new Set([2, 3]);
+      var Extra45  = new Set([1, 2, 3, 4, 5]);
+      var Disjoint = new Set([7, 8, 9]);
+      var Empty    = new Set([])
+      //
+      var WithItself   = functions.intersection(Exemplar, Exemplar);
```
Comment (Yomguithereal) [inline test/set.js:21]: Idem.

File: test/set.js [hunk 4]
```diff
@@ -11,23 +11,93 @@ describe('Set functions', function() {
...
+      var WithExtras   = functions.intersection(Exemplar, Extra45);
+      assert.deepStrictEqual(Array.from(WithExtras), [2, 3, 1]);
+      var WithDisjoint = functions.intersection(Exemplar, Disjoint);
+      assert.deepStrictEqual(Array.from(WithDisjoint), []);
+      var WithEmpty    = functions.intersection(Exemplar, Empty);
+      assert.deepStrictEqual(Array.from(WithEmpty), []);
+    });
 
-      var I = functions.intersection(A, B);
+    it('Makes no guarantees for element ordering.', function() {
```
Comment (Yomguithereal) [inline test/set.js:35]: drop the capitalized `M` :)

**Review by Yomguithereal — COMMENTED** [review]
Hello @mrflip, just a few nitpicks but overall looks good to me.

File: CHANGELOG.md (hunk 1 above)
Comment (mrflip) [reply CHANGELOG.md]: ✅

File: test/set.js (hunk 2 above)
Comment (mrflip) [reply test/set.js]: I updated all, including the pre-existing ones, to all be lowerCameled and more than one character

File: test/set.js (hunk 3 above)
Comment (mrflip) [reply test/set.js]: ✅

File: test/set.js (hunk 4 above)
Comment (mrflip) [reply test/set.js]: Fixed. In an act of overkill, because the jest eslinter enforces lowercase titles, I added the mocha linter to package.json. It does not; but it did catch some cases where there were arrow functions, which mocha doesn't like. I can remove or file that commit under its own PR if you prefer.

**Comment by mrflip** [comment]
I believe I've caught all the test style violations.
